### PR TITLE
Add API Gateway and Lambda

### PR DESCRIPTION
**Pull request by jonathonherbert**
_co-authored-by:_ @tjsilver

## What does this change?

Adds an API gateway and lambda to the CDK stack. The end goal is an endpoint which we can post telemetry events to, and a lambda service that pushes those events onto a Kinesis stream.

## How to test

Run `npm i; npm run test` in the `cdk` directory.

The tests should pass – it's just running the CDK to make sure there aren't any runtime errors.

Run `npm run synth`. Cloudformation should be successfully generated in the `cdk` folder.

## How can we measure success?

This code generates a stack that instantiates a correctly configured API gateway and lambda.

## Have we considered potential risks?

No risk, as nothing uses this service yet.

## Dev questions

 - ~Can we generate a certificate within this CDK, or must this happen in the AWS console?~ We can generate a cert in our region with this method, but it's preferable to have a global cert, which allows us to automatically provision a CDN. We've taken that approach for now.
 - ~Does this configuration look reasonable? (Neither author has much experience with CDK, or provisioning AWS services in general!)~ We'll find out on review 😁

### CODE REVIEW

**Comment by AWare** [comment]
I like this.

Might add a comment block after the params to break things up. 
Maybe drop the default readme from cdk and the tests though, it's just checking that it exits cleanly which I'd say is the job of whatever is running it and building the output.

**Comment by jonathonherbert** [comment]
@AWare good points both, will add.

File: README.md [hunk 1]
```diff
@@ -1 +1,4 @@
 # tools-telemetry-service
+
+A service to receive telemetry events and pass them on to a kinesis stream.
```
Comment (jennygrahamjones) [inline README.md:3]: Is there anything more specific that could be added here, or are we too early in development of this service to expand?

Comment (jonathonherbert) [reply README.md:3]: We could be more specific now! I'll add some more text. We're about to add code for the lambda and some interfaces, which will come with more information about behaviour/contracts etc.

**Comment by jennygrahamjones** [comment]
> Can we generate a certificate within this CDK, or must this happen in the AWS console?
Based on the stack, it looks like the answer to this is yes? 👍

**Comment by jonathonherbert** [comment]
@jennygrahamjones – yes to which? 😅

**Comment by jennygrahamjones** [comment]
> @jennygrahamjones – yes to which? 😅 

🤦🏻‍♂️ 

The first part of the sentence - whether you can create the certificate here

**Comment by jonathonherbert** [comment]
Discussed w/ @jennygrahamjones but for the record – this stack now creates its own certificate, which is validated via our DNS validation lambda.

**Comment by sihil** [comment]
The us-east-1 requirement will be because it's a global api gateway. If you make it a local api gateway then the certificate will be a local one I think. But then you won't get a CDN automatically in front of it. So 🤷‍♂️ .

**Comment by jonathonherbert** [comment]
@sihil ah, that explains that, thanks for the note. I'll leave it as is for now.

**Review by SHession — APPROVED** [review]

**Pull request merged**